src/h3/api/_api_template.py

```diff
@@ -114,12 +115,38 @@ def h3_to_geo_boundary(h, geo_json=False):
     def k_ring(h, k=1):
         mv = mvi.disk(_in_scalar(h), k)
 
-        return _out_collection(mv)
+        return _out_unordered(mv)
+
+    def hex_range(h, k=1):
+        mv = mvi.disk(_in_scalar(h), k)
+
+        return _out_unordered(mv)
 
     def hex_ring(h, k=1):
         mv = mvi.ring(_in_scalar(h), k)
 
-        return _out_collection(mv)
+        return _out_unordered(mv)
+
+    def hex_range_distances(h, K):
+        h = _in_scalar(h)
+
+        out = [
+            _out_unordered(mvi.ring(h, k))
+            for k in range(K + 1)
+        ]
+
+        return out
+
+    def hex_ranges(hexes, K):
+        out = {
+            h: hex_range_distances(h, K)
+            for h in hexes
+        }
+
+        return out
+
+    def k_ring_distances(h, K):
+        return hex_range_distances(h, K)
```

Comment:
Not sure about a `k_ring` function that calls hex_range, seems like a naming issue.

Reply:
Agreed. I've never gotten the distinction between `k_*` and `hex_*`,  especially with `*_ring` and `*_range`.  Do these things correspond to implementation details (around hitting pentagons)?

Seeing as most users don't even know that pentagons are a possibility, would it make sense to focus the "external" API towards "cell graph adjacency" and the "internal" functions towards the implementation details of pentagons and icosahedron faces?

Reply:
Added a comment to the code on this solution/choice here: https://github.com/uber/h3-py/pull/79/commits/05dd65178f3ced0e98ceb61fe7cd2cc765d9b22f